src/redisearch_rs/rqe_iterators/src/not_iterator.rs

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! Not iterator implementation
+
+use ffi::t_docId;
+use inverted_index::RSIndexResult;
+
+use crate::{
+    RQEIterator, RQEIteratorError, RQEValidateStatus, SkipToOutcome, maybe_empty::MaybeEmpty,
+};
+
+pub struct NotIterator<'index, I> {
+    child: MaybeEmpty<I>,
+    max_doc_id: t_docId,
+    current_id: t_docId,
```

Comment:
I would honestly remove `current_id` and assume `result` as the `current` (incl current id)... Prevents all the scenarios where you might prevent to keep in sync? Is anyway what I did with the optional iterator, makes life easier IMHO.